tests/WE2E/run_WE2E_tests.py

```diff
@@ -214,17 +212,28 @@ def run_we2e_tests(homedir, args) -> None:
         if args.compiler == "gnu":
             # 2D decomposition doesn't work with GNU compilers.  Deactivate 2D decomposition for GNU
             if 'task_run_post' in test_cfg:
-                test_cfg['task_run_post'].update({"NUMX": 1})
+                test_cfg.update_from({
+                    "task_run_post": {
+                        "upp": {
+                            "namelist": {
+                                "update_values": {
+                                    "nampgb": {
+                                        "numx": 1,
+                                    },
+                                },
+                            },
+                        },
+                    }
+                })
```

Comment:
I don't love this.

Reply:
Did `black` do that? What is it's line length set to? If I remove the commas, and `black -l 100` produces
```
                test_cfg.update_from(
                    {
                        "task_run_post": {
                            "upp": {"namelist": {"update_values": {"nampgb": {"numx": 1}}}}
                        }
                    }
                )
```
which seems better to me. Or with one intermediate variable
```
                upp = {"namelist": {"update_values": {"nampgb": {"numx": 1}}}}
                test_cfg.update_from({"task_run_post": {"upp": upp}})
```